### PR TITLE
Adding GCP to quickstart

### PR DESCRIPTION
**Pull request by R-Lawton**
### Verification
1.  `export MGC_BRANCH=gh-502 export  MGC_ACCOUNT=R-Lawton`
2.  run ` curl https://raw.githubusercontent.com/R-Lawton/multicluster-gateway-controller/gh-502/hack/quickstart-setup.sh | bash`
3. configmap, secret and managed zone should be created as normal but for gcp

in user input, if the wrong value is given for provider or value is not in the correct case the script will fail and exit If credential is not in JSON format same thing fail and exit

https://github.com/Kuadrant/multicluster-gateway-controller/issues/502

### CODE REVIEW

File: hack/quickstart-setup.sh [hunk 1]
```diff
@@ -16,21 +16,29 @@
...
 
-MGC_BRANCH=${MGC_BRANCH:="main"}
 
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.quickstartEnv)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.kindUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.cleanupUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.deployUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.startUtils)"
-source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/kuadrant/multicluster-gateway-controller/${MGC_BRANCH}/hack/.setupEnv)"
+source /dev/stdin <<< "$(curl -s https://raw.githubusercontent.com/${MGC_ACCOUNT}/multi-cluster-traffic-controller/${MGC_BRANCH}/hack/.quickstartEnv)"
```
Comment (jasonmadigan) [inline hack/quickstart-setup.sh:30]: nice, this will be handy

File: hack/.quickstartEnv [hunk 2]
```diff
@@ -39,4 +50,33 @@ requiredENV() {
     read MGC_ZONE_ROOT_DOMAIN </dev/tty
     echo "export MGC_ZONE_ROOT_DOMAIN for future executions of the script to skip this step"
     fi
-}
\ No newline at end of file
+  else 
+      if [[ -z "${GOOGLE}" ]]; then
+      echo "Enter either credentials created either by CLI or by service account (Please make sure the credentials provided are in JSON format)"
+      read GOOGLE </dev/tty
+        if ! jq -e . <<< "$GOOGLE" >/dev/null 2>&1; then
```
Comment (philbrookes) [inline hack/.quickstartEnv:57]: would it make sense to do this after the `if [[ -z "${GOOGLE}" ]]; then` has closed, so both the input $GOOGLE or the pre-exported $GOOGLE are both verified?

Comment (philbrookes) [reply hack/.quickstartEnv:57]: for example:
```
if [[ -z "${GOOGLE}" ]]; then
      echo "Enter either credentials created either by CLI or by service account (Please make sure the credentials provided are in JSON format)"
      read GOOGLE </dev/tty
      echo "export GOOGLE for future executions of the script to skip this step"
fi
if ! jq -e . <<< "$GOOGLE" >/dev/null 2>&1; then
  echo "Credentials provided is not in JSON format"
  exit 1
fi
```

**Comment by philbrookes** [comment]
I've run this locally, and got: 
```
Error: evalsymlink failure on '/tmp/kustomize-2918865925/config/quick-start/workload-cluster' : lstat /tmp/kustomize-2918865925/config/quick-start: no such file or directory
error: no objects passed to apply
Waiting for github.com/R-Lawton/multicluster-gateway-controller.git/config/quick-start/workload-cluster workload cluster config apply... Trying again in 5s
```

Repeatedly until it failed, looking into it locally.

**Comment by philbrookes** [comment]
> I've run this locally, and got:
> 
> ```
> Error: evalsymlink failure on '/tmp/kustomize-2918865925/config/quick-start/workload-cluster' : lstat /tmp/kustomize-2918865925/config/quick-start: no such file or directory
> error: no objects passed to apply
> Waiting for github.com/R-Lawton/multicluster-gateway-controller.git/config/quick-start/workload-cluster workload cluster config apply... Trying again in 5s
> ```
> 
> Repeatedly until it failed, looking into it locally.

it's worked the second time around, I assume I had a badly formed variable.

**Comment by philbrookes** [comment]
> > I've run this locally, and got:
> > ```
> > Error: evalsymlink failure on '/tmp/kustomize-2918865925/config/quick-start/workload-cluster' : lstat /tmp/kustomize-2918865925/config/quick-start: no such file or directory
> > error: no objects passed to apply
> > Waiting for github.com/R-Lawton/multicluster-gateway-controller.git/config/quick-start/workload-cluster workload cluster config apply... Trying again in 5s
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Repeatedly until it failed, looking into it locally.
> 
> comparing it to the one for the control-cluster, it appears to be missing the `?ref=gh-502` on the end, I'll try adding that

I think the change is required here: https://github.com/R-Lawton/multicluster-gateway-controller/blob/gh-502/hack/quickstart-setup.sh#L45

**Review by philbrookes — APPROVED** [review]
/lgtm

verified locally

**Pull request merged**